src/segments/unity.go

```diff
@@ -0,0 +1,104 @@
+package segments
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/regex"
+)
+
+type Unity struct {
+	props properties.Properties
+	env   platform.Environment
+
+	UnityVersion  string
+	CSharpVersion string
+}
+
+func (u *Unity) Enabled() bool {
+	unityVersion, err := u.GetUnityVersion()
+	if err != nil {
+		u.env.Error(err)
+		return false
+	}
+	if unityVersion == "" {
+		return false
+	}
+	u.UnityVersion = unityVersion
+
+	csharpVersion, err := u.GetCSharpVersion()
+	if err != nil {
+		u.env.Error(err)
+	}
+	u.CSharpVersion = csharpVersion
+
+	return true
+}
+
+func (u *Unity) GetUnityVersion() (version string, err error) {
+	versionFilePath := filepath.Join("ProjectSettings", "ProjectVersion.txt")
+	if !u.env.HasFiles(versionFilePath) {
+		return "", nil
```

Comment:
I would log this as it helps with debugging more explicitly.

```suggestion
        u.env.Debug("No ProjectVersion.txt file found")
		return "", nil
```

And I'm pretty sure this can be a return statement only as you have named variables.